contracts/contracts/poolBooster/PoolBoosterSwapxDouble.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IBribe } from "../interfaces/poolBooster/ISwapXAlgebraBribe.sol";
+import { IPoolBooster } from "../interfaces/poolBooster/IPoolBooster.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { StableMath } from "../utils/StableMath.sol";
+
+/**
+ * @title Pool booster for SwapX concentrated liquidity where 2 gauges are created for
+ *        every pool. Ichi vaults currently have such setup.
+ * @author Origin Protocol Inc
+ */
+contract PoolBoosterSwapxDouble is IPoolBooster {
+    using StableMath for uint256;
+
+    // @notice address of the Bribes.sol(Bribe) contract for the OS token side
+    IBribe public immutable bribeContractOS;
+    // @notice address of the  Bribes.sol(Bribe) contract for the other token in the pool
+    IBribe public immutable bribeContractOther;
+    // @notice address of the OS token
+    IERC20 public immutable osToken;
+    // @notice 1e18 denominated split between OS and Other bribe. E.g. 0.4e17 means 40% to OS
+    //         bribe contract and 60% to other bribe contract
+    uint256 public immutable split;
+
+    // @notice if balance under this amount the bribe action is skipped
+    uint256 public constant MIN_BRIBE_AMOUNT = 1e10;
+
+    constructor(
+        address _bribeContractOS,
+        address _bribeContractOther,
+        address _osToken,
+        uint256 _split
+    ) {
+        require(
+            _bribeContractOS != address(0),
+            "Invalid bribeContractOS address"
+        );
+        require(
+            _bribeContractOther != address(0),
+            "Invalid bribeContractOther address"
+        );
+        // expect it to be between 1% & 99%
+        require(_split > 1e16 && _split < 99e16, "Unexpected split amount");
+
+        bribeContractOS = IBribe(_bribeContractOS);
+        bribeContractOther = IBribe(_bribeContractOther);
+        osToken = IERC20(_osToken);
+        split = _split;
+    }
+
+    function bribe() external override {
+        uint256 balance = osToken.balanceOf(address(this));
+        // balance too small, do no bribes
+        if (balance < MIN_BRIBE_AMOUNT) {
+            return;
+        }
+
+        uint256 osBribeAmount = balance.mulTruncate(split);
+        // -1 to prevent possible rounding issues with OS token
+        uint256 otherBribeAmount = balance - osBribeAmount - 1;
```

Comment:
Do we even have this problem any more? :D :D

Reply:
I know we solved this for the contract type accounts or EOA's that have opted out of rebasing. Since those had the CPT set to 1e18 and internal credits to the actual balance. 

Not sure this is solved for accounts that participate in the yield?

Reply:
It's now pixel perfect for until the resolution runs out, at which point it will error towards more money in the source and more money in the destination on a transfer.

Reply:
Thanks for pointing this out. Addressed here: https://github.com/OriginProtocol/origin-dollar/pull/2400/commits/a3f537714a170276b5ca1ce3840b6a1c0a4d2bf7